pkg/router/template/router.go

```diff
@@ -171,13 +172,23 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 }
 
 func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
+	// IDhash contains an obfuscated internal IP address that is the value
+	// passed in the cookie. The IP address is made more difficult to extract
+	// by including other internal information in the hash.
 	if len(alias.PreferPort) == 0 {
+		for i := range svc.EndpointTable {
+			endpoint := svc.EndpointTable[i]
```

Comment:
Better to do this when we add an endpoint - see: `https://github.com/pecameron/origin/blob/bz1318796/pkg/router/template/router.go#L507`  rather than in here `endpointsForAlias`. 

Reason is `endpointsForAlias`  is called every time we handle the template (router config) and there would be multiple calls in the lifetime of a router (as endpoints/routes get added/removed/modified). It's not performant to calculate the IDHash for all the endpoints on every single call to `endpointsForAlias`.  
Just do it once when we add the endpoint. 

That way, you can also inline the function as it will be only used in a single place.